src/js/components/Pagination/Pagination.js

```diff
@@ -0,0 +1,202 @@
+
+    const getItemIndices = nextPage => {
+      const startIndex = step * (nextPage - 1);
+      const endIndex = startIndex + step;
+      return { startIndex, endIndex };
+    };
+
+    const handleClick = (event, nextPage) => {
+      event.persist();
+      const adjustedEvent = event;
+      adjustedEvent.page = nextPage;
+
+      // for controlled use cases, provide user with info on
+      // what range of indices should be displayed given the active page
+      const { startIndex, endIndex } = getItemIndices(nextPage);
+      adjustedEvent.startIndex = startIndex;
+      adjustedEvent.endIndex = endIndex;
+
+      setActivePage(event.page);
+
+      if (onChange) {
+        onChange(event);
+      }
+    };
+
+    const NextIcon = theme.pagination.icons.next;
+    const PreviousIcon = theme.pagination.icons.previous;
+    const iconColor = theme.pagination.icons.color;
+
+    const navProps = {
+      next: {
+        // https://a11y-style-guide.com/style-guide/section-navigation.html#kssref-navigation-pagination
+        'aria-disabled': activePage === totalPages ? 'true' : undefined,
+        disabled: activePage === totalPages,
+        icon: <NextIcon color={iconColor} />,
+        onClick: event => {
+          const nextPage = activePage + 1;
+          handleClick(event, nextPage);
+        },
+        label: undefined,
+      },
+      previous: {
+        'aria-disabled': activePage === 1 ? 'true' : undefined,
+        disabled: activePage === 1,
+        icon: <PreviousIcon color={iconColor} />,
+        onClick: event => {
+          const previousPage = activePage - 1;
+          handleClick(event, previousPage);
+        },
+        label: undefined,
+      },
+    };
+
+    let controls = [
+      'previous',
+      ...beginPages,
+      ...beginFlex, // either "..." or single page to bridge start + middle
+      ...middlePages,
+      ...endFlex, // either "..." or single page to bridge middle + end
+      ...endPages,
+      'next',
+    ];
+
+    /* Set props for each page index. Each page index should display a
+     * clickable index, control, or placeholder (e.g. ellipsis) indicating
+     * more pages are available.
+     */
+    controls = controls.map(control => {
+      return {
+        active: control === activePage,
+        a11yTitle:
+          typeof control === 'number'
+            ? `Go to page ${control}`
+            : `Go to ${control} page`,
+        // https://a11y-style-guide.com/style-guide/section-navigation.html#kssref-navigation-pagination
+        // https://www.w3.org/TR/wai-aria-1.1/#aria-current
+        'aria-current': control === activePage ? 'page' : undefined,
+        control,
+        onClick: event => {
+          handleClick(event, control);
+        },
+        separator: control === 'more-prev' || control === 'more-next',
+        ...navProps[control],
+      };
+    });
+
+    return (
+      <StyledPaginationContainer {...theme.pagination.container} {...rest}>
```

Comment:
Do we even need the container? Can we just have the Nav be the outermost component?

Reply:
The reason I had added it now was thinking if we'd ever want Pagination to have ability to internally control some sort of "showSummary" where user could pass in a summary message like "Showing 1-10 of 95 results". Mostly thinking of it from the perspective of use w DataTable/List where Pagination is controlled internally by those components. Felt that if that was something we may want to add in future, setting up the theming in this way would leave us flexibility for that.

Reply:
I'm generally reluctant to plan for potential future functionality too much. But, I think I can overcome that in this instance.

Reply:
Sounds good. I understand reluctance to do that in general.